pkg/cloudprovider/provider/aws/provider.go

```diff
@@ -752,6 +752,11 @@ func awsErrorToTerminalError(err error, msg string) error {
 			return prepareAndReturnError()
 		}
 		switch aerr.Code() {
+		case "InstanceLimitExceeded":
+			return cloudprovidererrors.TerminalError{
+				Reason:  v1alpha1.InsufficientResourcesMachineError,
+				Message: "You've reached the limit on the number of instances you can run concurrently. The limit depends on the instance type",
```

Comment:
Rather
`You've reached the AWS quota for number of instances of this type`

We should make it explicit that its about the AWS instance quote, as the termin quota occurs in k8s as well. Just to avoid confusion